src/core/server/saved_objects/migrations/core/opensearch_index.ts

```diff
@@ -68,6 +69,7 @@ export async function fetchInfo(
 
   const [indexName, indexInfo] = Object.entries(body)[0];
 
+  // @ts-expect-error @opensearch-project/opensearch IndexState.alias and IndexState.mappings should be required
```

Comment:
They should be required or they are required?

Reply:
assertIsSupportedIndex requires param type FullIndexInfo
```
export interface FullIndexInfo {
  aliases: { [name: string]: object };
  exists: boolean;
  indexName: string;
  mappings: IndexMapping;
}
```

indexInfo: opensearchtypes.IndicesIndexState
```
export interface IndicesIndexState {
  aliases?: Record<IndexName, IndicesAlias>
  mappings?: MappingTypeMapping
  settings: IndicesIndexSettings | IndicesIndexStatePrefixedSettings
}
```
{ ...indexInfo, exists: true, indexName } provides exists, indexName, and mappings
But aliase and mappings are optional in IndicesIndexState which is defined in the client.

**They are required by the type FullIndexInfo but the function assertIsSupportedIndex does not require.** First of all this assertIsSupportedIndex does not check alias. Second, for mapping, if mapping does not exist, this assertIsSupportedIndex will throw an error and a msg 

```
   `Index ${indexInfo.indexName} belongs to a version of OpenSearch Dashboards ` +
        `that cannot be automatically migrated. Reset it or use the X-Pack upgrade assistant.`
```

which is the checking purpose.